rustls/src/tls12/cipher.rs

```diff
@@ -99,6 +125,29 @@ impl Tls12AeadAlgorithm for Aes256Gcm {
     ) -> Box<dyn MessageEncrypter> {
         make_aesgcm_encrypter(enc_key, write_iv, explicit, &aead::AES_256_GCM)
     }
+
+    #[cfg(feature = "secret_extraction")]
+    fn extract_keys(&self, key: &[u8], iv: &[u8], explicit: &[u8]) -> ConnectionTrafficSecrets {
+        let key = {
```

Comment:
Can we deduplicate this, maybe using a function with some `const` parameters?

Reply:
I've reduced these to a handful of lines each; just the `ConnectionTrafficSecrets` item construction